test/unit/force_all_deps_direct/generator.bzl

```diff
@@ -39,6 +42,12 @@ EOF
         lib_hash = output_hash,
         extension = ".rlib",
     )
+    rust_metadata_name = "{prefix}{name}-{lib_hash}{extension}".format(
```

Comment:
This rule has a different purpose than testing pipelining, could we revert the changes here and clone the file under test/unit/pipelined_compilation and make the changes there?

Addidionally, could we have a rule like this one, that doesn't support pipelined compilation, and check that a 
rust_library_a -> custom_rule_without_compilation -> rust_library_b works?
We could have two different tests:
a) with enabled pipelining, the `custom_rule_without_compilation` action should still compile with the rust_library_b`.rmeta` input, even though `custom_rule_without_compilation` will not produce `.rmeta` output itself.
b) with enabled pipelining, the `rust_library_a` target will still compile successfully, by using the `custom_rule_without_compilation` `.rlib` output.

Reply:
Done.